### PR TITLE
Create react app

### PR DESCRIPTION
**Pull request by nahnahdev**
## Project Requirements

- [ ] Went through the React tutorial: [The beginner's guide to learning React in 2020](https://ibaslogic.com/react-tutorial-for-beginners/) step by step to build the project
- [ ] Working with React form and handling events
- [ ] How to implement CSS in the Reactjs app
- [ ] How to edit to dos items
- [ ] Persisting React state in local storage
- [ ] Getting started with React lifecycle methods
- [ ] Getting started with React hooks
- [ ] How to use SVG icons in React
- [ ] Routing with React Router
- [ ] How to add a hamburger menu in React
- [ ] Deploying React app to GitHub Pages.

### CODE REVIEW

File: src/functionBased/components/InputTodo.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    e.preventDefault();
+    const { addTodoProps } = props;
+    if (inputText.title.trim()) {
+      addTodoProps(inputText.title);
+      setInputText({
+        title: '',
+      });
+    } else {
+      alert('Please write item');
+    }
```
Comment (amon-cofie) [inline src/functionBased/components/InputTodo.js]: As per the JavaScript best practices, I suggest you avoid the use of alert() in your code. You can use an inline element like a span tag to display your message to your user

File: src/functionBased/components/InputTodo.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <form onSubmit={handleSubmit} className="form-container">
+      <input
+        type="text"
+        className="input-text"
+        placeholder="Add todo..."
+        value={inputText.title}
+        name="title"
+        onChange={onChange}
+      />
```
Comment (amon-cofie) [inline src/functionBased/components/InputTodo.js]: [OPTIONAL] It is a good idea to have the isRequired flag on form controls so as to ensure that a user doesn't submit the form with the input field empty

File: src/functionBased/components/TodoItem.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  useEffect(
+    () => () => {
+      console.log('Cleaning up...');
+    },
```
Comment (amon-cofie) [inline src/functionBased/components/TodoItem.js]: I also suggest you avoid the use of console.log() in your code per the JavaScript best practices.

**Review by amon-cofie — CHANGES_REQUESTED** [review]
Hi @NahnahAJ  :wave: 

You've done great work on your project :+1: :1st_place_medal: 
There are a few changes I would like to suggest but you're almost there :muscle: :rocket: 



## Highlights
- Correct use of Github Flow :green_circle: 
- No linter errors :green_circle: 
- Descriptive Pull Request :green_circle: 

## Required Changes :recycle: 
- [OPTIONAL] I suggest you make your commit messages imperatives. You can check [this link](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message) to learn how alter commit messages. I mark this suggestion as optional because it doesn't break your code and you have numerous commit messages so it would be inconvenient to have to change all of them but I highly recommend that you do as it is one of the [professional_repo_rules](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)

- Please check the comments under the review.

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. 

Cheers and Happy coding! :clap:

**Review by amon-cofie — APPROVED** [review]
Hi @NahnahAJ , 

## Status: APPROVED :green_circle: 

Your project is complete! Well done!! :1st_place_medal:   It's time to merge it :100: 
Congratulations! 🎉 



## Highlights

- All requested changes implemented :heavy_check_mark: 

Cheers and Happy coding!👏👏👏

File: src/functionBased/components/TodoItem.js (hunk 3 above)
Comment (nahnahdev) [reply src/functionBased/components/TodoItem.js]: Thanks a lot for your suggestions @amon-cofie . I will do well to implement them.

**Pull request merged**